napari/components/viewer_model.py

```diff
@@ -600,8 +498,60 @@ def add_multichannel(
                 blending=blending,
                 visible=visible,
             )
-            layers.append(layer)
-        return layers
+            self.add_layer(layer)
+            return layer
+        else:
+            n_images = data.shape[channel_axis]
+
+            name = ensure_iterable(name)
+
+            if blending is None:
+                blending = 'additive'
+
+            base_colormaps = [
+                'red',
+                'green',
+                'blue',
+                'cyan',
+                'yellow',
+                'magenta',
+            ]
+            if colormap is None:
+                colormap = itertools.cycle(base_colormaps)
+            else:
+                colormap = ensure_iterable(colormap)
+
+            # If one pair of clim values is passed then need to iterate them to
+            # all layers.
+            if contrast_limits is not None and not is_iterable(
+                contrast_limits[0]
+            ):
+                contrast_limits = itertools.repeat(contrast_limits)
+            else:
+                contrast_limits = ensure_iterable(contrast_limits)
+
+            layer_list = []
+            zipped_args = zip(range(n_images), colormap, contrast_limits, name)
+            for i, cmap, clims, name in zipped_args:
+                image = data.take(i, axis=channel_axis)
```

Comment:
This code probably breaks for pyramids... =\

Reply:
yup, fixed, thanks for your suggestion